test-network-function/lifecycle/suite.go

```diff
@@ -300,6 +300,19 @@ func shutdownTest(podNamespace, podName string) {
 	test.RunAndValidate()
 }
 
+func testNodeDrain(env *config.TestEnvironment, nodeName string) {
+	// We need to delete all Oc sessions because the drain operation is often deleting oauth-openshift pod
+	// This results in lost connectivity for oc sessions
+	env.ResetOc()
+	// drain node
+	drainNode(nodeName)
+	// Ensure the node is uncordoned before exiting the function
```

Comment:
still need the checking here so we know if pods get recreated properly, i.e. test pass or fail

Reply:
I'm not sure about that. waitForAllDeploymentsReady() is already checking if all pods are already recreated, and failing if they're not. So that status check was redundant, and we were never reaching it in case of a timeout.